pkg/apis/workflow/v1alpha1/cron_workflow_types.go

```diff
@@ -55,6 +55,8 @@ type CronWorkflowSpec struct {
 	Timezone string `json:"timezone,omitempty" protobuf:"bytes,8,opt,name=timezone"`
 	// WorkflowMetadata contains some metadata of the workflow to be run
 	WorkflowMetadata *metav1.ObjectMeta `json:"workflowMetadata,omitempty" protobuf:"bytes,9,opt,name=workflowMeta"`
+	// SuspendAfterFailedJobsLimit is the number of failed jobs allowed before a CronWorkflow is suspended automatically
+	SuspendAfterFailedJobsLimit *int32 `json:"suspendFailedJobsLimit,omitempty" protobuf:"varint,10,opt,name=suspendFailedJobsLimit"`
```

Comment:
Perhaps the terminology here should be closer to "Unsuccessful Submission" rather than "Failed Job". If we are able to successfully submit a workflow, but it fails on its own, it's not of interest here.